tiledb/sm/consolidator/consolidator.cc

```diff
@@ -146,21 +146,21 @@ void Consolidator::array_consolidate(
     throw ConsolidatorException("Cannot consolidate array; Invalid URI");
   }
 
-  // Check if array exists
-  ObjectType obj_type;
-  throw_if_not_ok(
-      object_type(storage_manager->resources(), array_uri, &obj_type));
-
-  if (obj_type != ObjectType::ARRAY) {
-    throw ConsolidatorException(
-        "Cannot consolidate array; Array does not exist");
-  }
-
   if (array_uri.is_tiledb()) {
     throw_if_not_ok(
         storage_manager->resources().rest_client()->post_consolidation_to_rest(
             array_uri, config));
   } else {
+    // Check if array exists
+    ObjectType obj_type;
+    throw_if_not_ok(
+        object_type(storage_manager->resources(), array_uri, &obj_type));
```

Comment:
moved this in the `else` case so that it's not executed for remote arrays and cause 2 extra REST requests (`is_group` and `is_array`)